src/engagement_db_to_analysis/data_wrangling_functions.py

```diff
@@ -0,0 +1,81 @@
+from core_data_modules.util import TimeUtils
+
+from core_data_modules.data_models.code_scheme import CodeTypes
+from core_data_modules.cleaners.cleaning_utils import CleaningUtils
+from core_data_modules.traced_data import Metadata
+from core_data_modules.logging import Logger
+
+
+log = Logger(__name__)
+
+def _impute_age_category(user, messages_traced_data, analysis_dataset_configs):
+    """
+    Imputes age category for age dataset messages.
+
+    :param user: Identifier of user running the pipeline.
+    :type user: str
+    :param messages_traced_data: Messages TracedData objects to impute age_category.
+    :type messages_traced_data: list of TracedData
+    :param analysis_dataset_config: Analysis dataset configuration in pipeline configuration module.
+    :type analysis_dataset_config: pipeline_config.analysis_configs.dataset_configurations
+    """
+
+    log.info("Imputing age categories for age dataset messages...")
+    # Get the configurations for age and age_category analysis datasets
+    updated_messages_traced_data = []
+    age_cc = None
+    age_category_cc = None
+    age_engagement_db_datasets = None
+    for analysis_dataset_config in analysis_dataset_configs:
+        for coding_config in analysis_dataset_config.coding_configs:
+            if coding_config.analysis_dataset == "age":
```

Comment:
I'm not sure about this because it means we always need to call these datasets age and age_category, which might not be true and it's not clear when writing a configuration file that you need to do this.

I think I see the problem though, which is that the age_categorisation needs to know where to look to find the age dataset - maybe change the configuration to something like
age_category=AgeCategoryConfiguration(age_dataset="age", [categories=...])

Reply:
It's not just that we also need to loop through age messages only, and to determine this we have to check for an age config property in each message. The alternative here is to define constants e.g AGE_CATEGORY = "age_category" in configuration.py that we can then use in the scripts in place of strings.